buildSrc/build.gradle

```diff
@@ -130,7 +130,7 @@ dependencies {
   testFixturesApi gradleTestKit()
   testImplementation 'org.wiremock:wiremock-standalone:3.6.0'
   testImplementation "org.mockito:mockito-core:${props.getProperty('mockito')}"
-  integTestImplementation('org.spockframework:spock-core:2.4-M6-groovy-4.0') {
+  integTestImplementation('org.spockframework:spock-core:2.3-groovy-3.0') {
```

Comment:
#19329 also added JVM args change:

```
jvmArgs += ["--add-opens", "java.base/java.lang=ALL-UNNAMED"]
```

I am wondering if we really need that?

Reply:
By the way, I intentionally didn't revert all of #19329 because it includes a bunch of changes to stop using deprecated Gradle APIs which still work with Gradle 8.

Reply:
> It removes a ton of warnings during build. Some future version of the JDK will fail without it.

Got it, thanks!

> By the way, I intentionally didn't revert all of https://github.com/opensearch-project/OpenSearch/pull/19329

Yeah, I also noticed that. Just was slightly confused about this jvmArgs change. Thanks for the explanation!